src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -386,7 +615,165 @@ public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilde
             );
         }
     }
+    public static class RenderedPolyface extends RenderedShape<ShapeDispatcher.Polyface>
+    {
+        public RenderedPolyface(Minecraft client, ShapeDispatcher.ExpiringShape shape)
+        {
+            super(client, (ShapeDispatcher.Polyface)shape);
+        }
+        @Override
+        public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, double cx, double cy, double cz, float partialTick)
+        {       
+            if(shape.fa==0){return;}
+            if (shape.needf3b && !client.getEntityRenderDispatcher().shouldRenderHitBoxes()){
```

Comment:
I higly suggest you move the F3b option up to the any shape category and add that as an extra condition to render any shape.

Reply:
🆗 i moved it to its father class